radixdlt-java/src/main/java/com/radixdlt/client/core/network/actions/FetchAtomsObservationAction.java

```diff
@@ -0,0 +1,66 @@
+package com.radixdlt.client.core.network.actions;
+
+import com.radixdlt.client.atommodel.accounts.RadixAddress;
+import com.radixdlt.client.core.atoms.AtomObservation;
+import com.radixdlt.client.core.network.RadixNode;
+import java.util.Objects;
+
+/**
+ * A dispatchable fetch atoms action which represents an atom observed event from a specific node for an atom fetch flow.
+ */
+public final class FetchAtomsObservationAction implements FetchAtomsAction {
+	private final String uuid;
```

Comment:
No inheritance please! :) But really inheritance via `extend` should only be used for exceptional cases as most things should be done via composition as it avoids coupling a lot of nasty issues down the line. The "extra" code that comes from re-implementing an interface is MUCH cheaper in complexity management compared to using a common base class.  https://codeburst.io/inheritance-is-evil-stop-using-it-6c4f1caf5117 summarizes my view.

Reply:
I'm gonna keep pushing this because it's very important :)

The idea that inheritance should most usually be avoided (especially for java) isn't just a matter of preference but is a widely accepted view:

https://hackernoon.com/inheritance-based-on-internal-structure-is-evil-7474cc8e64dc

https://www.quora.com/Is-inheritance-bad-practice-in-OOP-Many-places-that-teach-design-patterns-say-to-opt-for-composition-over-inheritance-but-what-about-when-multiple-classes-share-logic-from-an-abstract-class-such-as-in-the-Template-Method-design-pattern

That's the point of composition, every implementation can store it how it wants. The reason to store it in every single implementation is for decoupling which keeps every implementation modular and independent from each other. Once you add a base class, now every class is dependent on the same class and any modifications to the base class will have repercussions on all classes extending it which is a mess when code gets more and more complex.

Reply:
> Dogs don’t inherit from Trees just because you want a bark() method. Keep that in mind.
Haha. Nice. 

Interesting, you got me convinced for the most part. I like composition, that's why I wanted to have Checks in the validation logic. And then again, I also implemented some of the validation structure through inheritance. How would that be improved in a composable manner? The subclasses in my validation implementation are basically just typedefs for convenience. 

We should have a discussion what we want Core to look like in a perfect world, so that we can prioritise our refactors and feature implementations properly. 

What about stuff like Quarks, Particles? I don't see any issues with having a base Particle and Quark class and more specific implementations.

Reply:
Also I love how the answer right below the one you linked me is all for inheritance again. Widely accepted indeed ;)